colorama/ansitowin32.py

```diff
@@ -87,14 +87,24 @@ def __init__(self, wrapped, convert=None, strip=None, autoreset=False):
         # to support the ANSI codes.
         conversion_supported = on_windows and winapi_test()
 
+        # if the stream is not closed, check if it's a TTY
+        not_closed = not self.stream.closed
+        isatty = not_closed and self.stream.isatty()
+
+        # check if the FORCE_COLOR environment variable has been set and NO_COLOR has not been set.
+        force_color = 'FORCE_COLOR' in os.environ.keys()
+        no_color = 'NO_COLOR' in os.environ.keys()
+
+        color_allowed = (isatty or force_color) and not no_color
```

Comment:
Possibly clearer as:
```
if  'NO_COLOR' in os.environ:
    color_allowed = False
elif 'FORCE_COLOR' in os.environ:  # overriden by NO_COLOR
    color_allowed = True
else:
    color_allowed = isatty
```

Reply:
Is this clearer? Just seems more verbose to me. But if changing it would get this PR over the line, I'm happy to make the change.

Reply:
It makes it clear that there's a priority order of things to look at. Clearness is certainly not measured by character count alone ;).

I'm not part of this project though, so my review only matters if a maintainer says it does. I'm here because I received a PR for a project I maintain that added `NO_COLOR`, and wanted to see how it should interact with `FORCE_COLOR`.

Reply:
Hey thanks for taking the time to review the PR.

And ok... you've talked me into it. I made the change. Your `'FORCE_COLOR' in os.environ` is superior to my 'FORCE_COLOR' in os.environ.keys()`.

Hopefully we can get this merged as soon as possible. Right now this PR feels like it's stuck in bike-shed hell.

Reply:
I hear you. Apologies, that's at least partly my fault. I'll find time today to progress it, hopefully to merge. Thank you for all your work, it really is very much appreciated.

Reply:
No worries I quite understand - I'm just keen to get this closed out, and I think it would save all our time if we could reduce the number of round trips, so if there's anything you want me to do, just lay it out - or even feel free to add your own patches, or amend mine.

It sounds like the concept of this PR is acceptable. In which case it's just a question of whether the implementation is sufficient in quality. The code has gone through a few iterations now, so it seems pretty good to me, but let me know if there are any desired fixes.